CleverTapSDK/CTInAppDisplayViewController.m

```diff
@@ -129,18 +197,66 @@ - (void)hide:(BOOL)animated {
 }
 
 - (void)initializeWindowOfClass:(Class)windowClass animated:(BOOL)animated {
+    float aspectRatio = self.notification.aspectRatio;
+    float percent = self.notification.widthPercent;
+    
     if (@available(iOS 13, tvOS 13.0, *)) {
         NSSet *connectedScenes = [CTUIUtils getSharedApplication].connectedScenes;
         for (UIScene *scene in connectedScenes) {
             if (scene.activationState == UISceneActivationStateForegroundActive && [scene isKindOfClass:[UIWindowScene class]]) {
                 UIWindowScene *windowScene = (UIWindowScene *)scene;
-                self.window = [[windowClass alloc] initWithFrame:
-                               windowScene.coordinateSpace.bounds];
+                
+                float safeAreaTop = windowScene.windows.firstObject.safeAreaInsets.top;
```

Comment:
@Sonal-Kachare  i see some logic is duplicated such as this and the following lines. Can you move them to a common location if possible?

Reply:
I've tried to reduce code duplication